### DISCUSSION TITLE
Milk-V Pioneer support

### DISCUSSION THREAD

**Discussion by kth5** [kth5/archpower#68] [Announcements]
The Box has arrived today an work is underway:
![image](https://github.com/kth5/archpower/assets/1784992/2b972a22-d0d3-4bfa-8e42-cbbdd68c792c)

**Comment by 197788** [comment]
You have the documentation on how this was achieved?

**Reply by kth5** [reply]
I'm working on extending the sd-creator tool which is currently only supporting the visionfive 2 sbc to work here as well. Basiclaly I  just took the fedora image provided and swapped out the rootfs for an ArchPOWER one and modified extlinux.conf to use our kernel instead.

Right now, building the firmware bits is still a bit of a mystery to me as the documentation is pretty lacking.

If you'd I can provide a simple 8GiB image to get you started but it won't be release quality as its still using Fedora's boot glue.

**Reply by 197788** [reply]
I hear you...thanks

**Comment by kth5** [comment]
So since my Pioneer basically died (a tad over a year old) with its PCI-e slots being shot since the 4th month (distributor and Milk-V were no help) in and it now starting to just suck in general. Additionallty the vendor documentation is basically useless for purposes of rolling a distro for this. I am afraid this is on hold again until further notice (which likely never will come).

Thus far Starfive seems to produce the most stable and well documented RISC-V hardware so expect that work to go towards their tech.
